src/runtime/virtcontainers/sandbox.go

```diff
@@ -825,6 +834,106 @@ func (s *Sandbox) coldOrHotPlugVFIO(sandboxConfig *SandboxConfig) (bool, error)
 	return coldPlugVFIO, nil
 }
 
+func (s *Sandbox) prepareInitdataMount(config *HypervisorConfig) error {
+	if len(config.Initdata) == 0 {
+		s.Logger().Info("No initdata provided. Skip prepare loop device")
+		return nil
+	}
+
+	s.Logger().Info("Start to prepare initdata")
+	initdataWorkdir := filepath.Join(string(filepath.Separator), "/run/initdata", s.id)
+	initdataImagePath := filepath.Join(string(filepath.Separator), initdataWorkdir, "data.img")
```

Comment:
```suggestion
	initdataWorkdir := filepath.Join("/run/initdata", s.id)
	initdataImagePath := filepath.Join(initdataWorkdir, "data.img")
```

Reply:
This is similar to https://github.com/kata-containers/kata-containers/blob/4f41989a6a817b384aa2ced2bfb64dd954f62403/src/runtime/pkg/containerd-shim-v2/shim_management.go#L345 for cross-platform compatibility

Reply:
`Join()` uses OS specific separator. At least `string(filepath.Separator), "/run/initdata",` is redundant because the result is `Clean()`ed so you get `/run/initdata` back. Is there a need to prefix these paths with `/`?

Reply:
Sounds reasonable, let me fix this.